content/learn/book/welcome/apps/_index.md

```diff
@@ -0,0 +1,81 @@
+The process is straightforward: we first create a new {{rust_type(type="struct" crate="bevy" mod = "app" name="App" no_mod = "true")}}.
+Then, we add a simple system, which prints "Hello, Bevy!" when it is run.
+Finally once we're done configuring the app, we call {{rust_type(type="struct" crate="bevy" mod = "app" name="App" no_mod = "true")}} to actually make our app *do things*.
+
+```rust
+use bevy::prelude::*;
+
+fn main(){
+    App::new()
+        .add_system(hello)
+        .run();
+}
+
+fn hello(){
+    println!("Hello, Bevy!")
+}
+```
+
+## What makes an App?
+
+So, what sort of data does our  {{rust_type(type="struct" crate="bevy" mod = "app" name="App" no_mod = "true")}} really store?
+Looking at the docs linked, we find three fields: `world`, `schedule` and `runner`.
+The `world` field stores all of our game's data, the `schedule` holds the systems that operate on this data (and the order in which they do so) and the `runner` interprets the schedule to control the broad execution strategy.
+You can read more about these by exploring the reference documentation linked just above.
+
+Generally, you'll be operating at a more granular level than these basic primitives: controlling data in terms of specific resources or components and adding systems to an existing schedule.
+To do so, customize your own  {{rust_type(type="struct" crate="bevy" mod = "app" name="App" no_mod = "true")}} by chaining its methods with the [builder pattern](https://doc.rust-lang.org/1.0.0/style/ownership/builders.html).
+The most basic tools are:
+
+  1. Initializing resources in the {{rust_type(type="struct" crate="bevy" mod = "ecs/world" name="World" no_mod = "true")}} to store global data.
+  2. Adding systems to our {{rust_type(type="struct" crate="bevy" mod = "ecs/schedule" name="Schedule" no_mod = "true")}} to perform logic.
+  3. Importing other blocks of {{rust_type(type="struct" crate="bevy" mod = "app" name="App" no_mod = "true")}}-modifying code using plugins.
+Let's write a very simple demo that shows how those work.
+
+```rust
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        // Plugins are App code that was written elsewhere,
+        // imported as a single unit for organization and clarity
+        .add_plugins(DefaultPlugins)
+        // Resources are global singleton data stored in the `World`
+        .insert_resource(Fibonacci { a: 1, b: 1 })
+        // Systems run every pass of the game loop and perform logic
+        .add_system(fibonacci_sum.system())
+        .add_system(report_fibonacci.system())
+        .run();
+}
+
+// A simple data structure to store the two current fibonacci numbers
+struct Fibonacci {
+    a: u64,
+    b: u64,
+}
+
+// This system requires mutable access to our Fibonacci resource,
+// so we add `ResMut` to its function parameters
+fn fibonacci_sum(mut fibonacci: ResMut<Fibonacci>) {
+    // This crashes fairly quickly, as we overflow our u64 data storage
```

Comment:
Should we put the code in to stop the crash? As a beginner I would be hesitant to run something that crashes. Perhaps add a line with checks then comment that with something like "Learning exercise: What would happen if we removed this line?"

That may be too "tutorial" but just an idea.

Reply:
Yeah I'd rather just avoid having an example that requires special casing, crashes, or warning comments in the first place. Fibonacci (while extremely common in programming and math circles) might not be the best choice for some types of users. Maybe we should pick something a little more "generic" and approachable. Maybe something like a "greeting message" logger that prints a greeting message from a `struct GreetingMessage { message: String }` resource?